module-gui/gui/widgets/text/core/lines/Lines.cpp

```diff
@@ -33,13 +34,27 @@ namespace gui
         }
     }
 
-    auto Lines::draw(BlockCursor &drawCursor, Length w, Length h, Position lineYPosition, Position lineXPosition)
+    auto Lines::draw(
+        BlockCursor &drawCursor, Length w, Length h, Position lineYPosition, Position lineXPosition, TextType drawType)
         -> void
+    {
+        drawType == TextType::MultiLine ? drawMultiLine(drawCursor, w, h, lineYPosition, lineXPosition)
+                                        : drawSingleLine(drawCursor, w, h, lineYPosition, lineXPosition);
```

Comment:
again - why not make Lines a base and do behaviour specific things in a specified class? :) 
this would be logical: single line cursor -> triggers single line behaviour, multiline cursor -> triggers multiline behaviour

Reply:
There are many dependencies in both cursors and text on lines object. Hard to do for now.